### PR TITLE
[Merged by Bors] - feat: port number_theory.legendre_symbol.norm_num to Tactic.NormNum.LegendreSymbol

### PR DESCRIPTION
**Pull request by Komyyy**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/NormNum/LegendreSymbol.lean [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+      show (er : Q(ℤ)) × Q(jacobiSymNat 1 $eb = $er) from
+        ⟨mkRawIntLit 1, q(jacobiSymNat.one_left $eb)⟩
+    | a =>
+      match a % 2 with
+      | 0 =>
+        match a % 4 with
+        | 0 =>
+          have ha : Q(Nat.mod $ea 4 = 0) := (q(Eq.refl 0) : Expr)
+          have hb : Q(Nat.mod $eb 2 = 1) := (q(Eq.refl 1) : Expr)
+          have c := a / 4
```
Comment (semorrison) [inline Mathlib/Tactic/NormNum/LegendreSymbol.lean:278]: You could inline quite a few of these haves into the (sole) place the definition is used. If you think it hurts readability I guess we can leave it.

**Comment by semorrison** [comment]
Looks very nice!

bors d+

File: Mathlib/Tactic/NormNum/LegendreSymbol.lean [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+      match ea.natLit! with
+      | 0 =>
+        have hb : Q(Nat.beq ($eb / 2) 0 = false) := (q(Eq.refl false) : Expr)
+        show (er : Q(ℤ)) × Q(jacobiSymNat 0 $eb = $er) from
+          ⟨mkRawIntLit 0, q(jacobiSymNat.zero_left $eb $hb)⟩
+      | 1 =>
+        show (er : Q(ℤ)) × Q(jacobiSymNat 1 $eb = $er) from
+          ⟨mkRawIntLit 1, q(jacobiSymNat.one_left $eb)⟩
+      | a =>
+        match a % 0 with
```
Comment (MichaelStollBayreuth) [inline Mathlib/Tactic/NormNum/LegendreSymbol.lean:349]: This looks incorrect. I guess it should be `a % 2` instead of `a % 0`.

Comment (Komyyy) [reply Mathlib/Tactic/NormNum/LegendreSymbol.lean:349]: Oh, that's wrong, thank you.
I'll create the PR after porting `Traversable` deriving handler.

Comment (Komyyy) [reply Mathlib/Tactic/NormNum/LegendreSymbol.lean:349]: #5719

**Pull request closed without merging**